docs/Blog-Post.md

```diff
@@ -0,0 +1,525 @@
+and [DeepSpeed](https://github.com/microsoft/DeepSpeed).
+
+In the JAX community, there are a number of libraries popping up. Google has released [T5X](https://github.com/google-research/t5x)
+and [MaxText](https://github.com/google/maxtext). There are also a number of independent libraries, including [EasyLM](https://github.com/young-geng/EasyLM)
+and [JAXSeq](https://github.com/Sea-Snell/JAXSeq), both of which are based on [Flax](https://github.com/google/flax/)
+and modified libraries from [Hugging Face Transformers](https://github.com/huggingface/transformers/). Salesforce has
+released the Haiku-based [JAXformer](https://github.com/salesforce/jaxformer). Previously, Eleuther AI released
+[mesh-transformer-jax](https://github.com/kingoflolz/mesh-transformer-jax/), though it is mostly unmaintained now and uses
+older, quasi-deprecated JAX APIs for distributed training.
+
+## A New Codebase for Foundation Model Training
+
+Despite the wide array of existing frameworks, when we started, we found that none of them fully addressed our needs.
+At CRFM, we focused on three fundamental goals:
+
+* **Legibility and Composability**: We prioritized writing code that is easy to read, understand, and compose.
+* **Reproducibility and Resilience**: We emphasized the ability to reproduce results *exactly*, even in the face of preemption and restarts from checkpoints.
+* **Scalability and Efficiency**: We aimed to fully utilize the Cloud TPU resources from Google's TPU Research Cloud program, as well as our own GPU resources, without compromising efficiency.
+
+We chose [JAX](https://github.com/google/jax/) as our framework because it is a powerful, flexible, and performant,
+and offers strong reproducibility guarantees. JAX also works well on TPUs, while we found that PyTorch support was still uneven.
+JAX is also a natural choice because it allows you to focus on the "what" of your code, and not on the "how": details of
+partitioning and communication can be left to the XLA compiler. Finally, JAX makes reproducibility easy, since it uses
+bitwise deterministic PRNGs by default, with careful control over the PRNG state.
+
+However, JAX is a low-level framework, and we found that, by itself, it did not provide the legibility that we wanted.
+
+For us, legibility is a top concern. We therefore created two new libraries: **Haliax** and **Levanter**. Haliax is a
+named tensor library that focuses on improving the legibility and compositionality of deep learning code while still
+maintaining efficiency and scalability. Levanter is a library for training foundation models built on top of Haliax. In
+addition to the goals of legibility, efficiency, and scalability, Levanter further strives for bitwise reproducibility,
+meaning that the same code with the same data will produce the exact same result, even in the presence of preemption and
+restarts.
+
+# Haliax: Legibility via Named Tensors
```

Comment:
Can we name this "Desideratum 1 (legibility): named tensors via Haliax" or "Legibility: named tensors via Haliax"?